elastica/wrappers/damping.py

```diff
@@ -0,0 +1,156 @@
+__doc__ = """
+Damping
+-----------
+
+Provides the damper interface to apply damping
+on the rods. (see `dissipation.py`).
+
+"""
+
+from elastica.dissipation import DamperBase
+
+
+class Damping:
+    """
+    The Damping class is a wrapper for applying damping
+    on rod-like objects, the simulator class must be derived from
+    Damping class.
+
+        Attributes
+        ----------
+        _dampers: list
+            List of damper classes defined for rod-like objects.
+    """
+
+    def __init__(self):
+        self._dampers = []
+        super(Damping, self).__init__()
+        self._feature_group_constrain_rates.append(self._dampen_rates)
+        self._feature_group_finalize.append(self._finalize_dampers)
+
+    def dampen(self, system):
+        """
+        This method applies damping on relevant user-defined
+        system or rod-like object. You must input the system or rod-like
+        object that you want to apply damping on.
+
+        Parameters
+        ----------
+        system: object
+            System is a rod-like object.
+
+        Returns
+        -------
+
+        """
+        sys_idx = self._get_sys_idx_if_valid(system)
+
+        # Create _Damper object, cache it and return to user
+        _damper = _Damper(sys_idx)
+        self._dampers.append(_damper)
+
+        return _damper
+
+    def _finalize_dampers(self):
+        # From stored _Damping objects, instantiate the dissipation/damping
+        # inplace : https://stackoverflow.com/a/1208792
+
+        self._dampers[:] = [
+            (damper.id(), damper(self._systems[damper.id()]))
+            for damper in self._dampers
+        ]
+
+        # Sort from lowest id to highest id for potentially better memory access
+        # _dampers contains list of tuples. First element of tuple is rod number and
+        # following elements are the type of damping.
+        # Thus using lambda we iterate over the list of tuples and use rod number (x[0])
+        # to sort dampers.
+        self._dampers.sort(key=lambda x: x[0])
+
+    def _dampen_rates(self, time, *args, **kwargs):
+        for sys_id, damper in self._dampers:
+            damper.dampen_rates(self._systems[sys_id], time, *args, **kwargs)
+
+
+class _Damper:
```

Comment:
Rename to `_Damping`? Its suppose to be private class wrapper for `Damping`.

Reply:
The reason for that is the Damping wrapper actually holds a list of the "dampers" initialized through the "_damper" class. Hence the naming. I'm not too stuck up on this and arman can change this if he feels like.

Reply:
Yeah, lets keep it consistent with other naming. Im looking into the code

Reply:
Okay, if this and the other changes are minor can you please take care of them?

Reply:
I think `_damper` make sense and consistent with other wrappers the for example `Forcing`. The main wrapper that calls all external force classes is called `Forcing` and individual external force objects are called `_ExtForceTorque` . Similarly we name wrapper that calls all `damper` classes as `Damping` and individual damper objects as `_Dampers`.